KSR_002/KSR_002.md

```diff
@@ -0,0 +1,282 @@
+- syscalls: e.g. for x86 [entry_SYSCALL_64](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/x86/entry/entry_64.S?h=v5.14#n87);
+- explicit invocation of the scheduler __schedule();
+- Interrupt and Exception handling
+- Signals being sent between processes (<mark > **TODO:** </mark> investigate further what can happen here)  
+context_switch() show user space mm being borrowed by kernel threads upon context switch:  
+<mark > **TODO:** </mark> understand why this is required and if we can use a dummy mm instead of borrowing a real user space mm  
+<mark > **TODO: (possibly in a different investigation thread)** </mark>: investigate the role of membarrier_switch_mm() when switching to user thread
+
+
+## d. 
+The Kernel subsystems and the HW supporting the integrity of processes address spaces upon creation and at runtime shall be properly configured at boot time. This is responsibility of start_kernel(); <mark > **TODO:** </mark> identify the subsystems of the Kernel that following the start_kernel() are invoked to meet this requirement  
+<mark > **TODO:** </mark> Investigate if init user space processes (e.g. systemd) can impact the capability/functionalities of Kernel subsystems responsible for guaranteeing the integrity of process address spaces during creation or runtime.
+
+
+# NOTE: 
+
+**Previous material relevant to this topic:**
+
+
+- [Safety in processes CPU execution state](https://linuxplumbersconf.org/event/7/contributions/698/): focused on the integrity of the CPU execution state.  
+Mainly focused on problem c. above. This session also mentions HW faults (that are out of scope WRT to our investigation). About SW Faults (slide10):
+    - Data corruption was skipped
+    - Task switching: how the process state is saved and restored
+    - Bad Kernel Code: TBD?
+- Codethink Mitigations:
+    - Shadowing the Thread Control Block (is it [https://elixir.bootlin.com/linux/v5.14.9/source/include/linux/sched.h#L661](https://elixir.bootlin.com/linux/v5.14.9/source/include/linux/sched.h#L661) ?)
+    - Wrapping syscall and other entry points 
+
+    - [https://www.codethink.co.uk/articles/2020/investigating-kernel-user-space-access/](https://www.codethink.co.uk/articles/2020/investigating-kernel-user-space-access/):  
+    - Focuses on mechanisms preventing corruption of user space process address spaces from kernel threads. This is in scope of point a. and c. above.
+    It considers both static and runtime features.
+    From a runtime point of view the considered features are:
+
+        - [Kernel Self Protection](https://www.kernel.org/doc/html/latest/security/self-protection.html)
+        - [Supervisor Mode Execution Prevention](https://en.wikipedia.org/wiki/Control_register#SMEP) (SMEP)
+        - [Supervisor Mode Access Prevention](https://en.wikipedia.org/wiki/Supervisor_Mode_Access_Prevention) (SMAP). 
+
+
+**Safe Process Context Protection:** Prevent the silent corruption of a safe process context.
```

Comment:
Following the discussion in the WG meeting we should move all the text from this line below into a separate MD file so that later on it would be easier to port it to a different WG and remove it from this github